src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -291,6 +291,14 @@ internal static class MethodGenerator
                 }
             }
 
+            if (destination is CodeGenerationDestination.CompilationUnit)
+            {
+                if (method.IsStatic)
+                {
+                    tokens.Add(SyntaxFactory.Token(SyntaxKind.StaticKeyword));
```

Comment:
this feels odd.  shouldn't this always be added if hte method is static?  to the point that i think we could just have this check before everything else, and unilaterally add 'static' (regardless oc ompilation-unit/type/interface destinatino).  thoughts?

Reply:
hrmm... i see.  there's an issue of wanting to avoid things like accessibility.

Reply:
i think this should move up to line 281 and become an 'else' of htat condition.